File: 2021/day01/python/part1.py
#!usr/bin/env python

import sys

def main(input):
    count = 0
    for i in range(0, len(input) - 1):
        if input[i] < input[i + 1]:
            count = count + 1
    return count

if __name__ == '__main__':
    fo = open(sys.argv[1], 'r')
    lines = fo.readlines()
    report = [int(line) for line in lines]
    print(main(report))
    fo.close()

File: 2021/day01/c/part1.c
#include <stdlib.h>
#include <stdio.h>
#include <string.h>

#define MAX_LINE_LENGTH 80

int solve(int *input, int c){
	int i;
	int count = 0;

	for (i = 0; i < c - 1; i++){
		if (input[i] < input[i + 1]){
			count++;
		}
	}

	return count;
}

int main(int argc, char **argv){
	FILE *fo;
	char line[MAX_LINE_LENGTH] = {0};
	int *lines;
	char c;
	int lineCount = 0;
	int i = 0;

	fo = fopen(argv[1], "r");

	for (c = getc(fo); c != EOF; c = getc(fo)){
		if (c == '\n') lineCount++;
	}

	lines = malloc(sizeof(int) * lineCount);
	rewind(fo);

	while(fgets(line, MAX_LINE_LENGTH, fo)){
		lines[i++] = atoi(line);
	}

	printf("%d\n", solve(lines, lineCount));
	
	free(lines);
	fclose(fo);
}

